examples/ip_tracker/Application.cpp

```diff
@@ -0,0 +1,336 @@
+    {
+        return "WARN";
+    }
+    else if (level == wolkabout::LogLevel::ERROR)
+    {
+        return "ERROR";
+    }
+    else if (level == wolkabout::LogLevel::DEBUG)
+    {
+        return "DEBUG";
+    }
+    else if (level == wolkabout::LogLevel::TRACE)
+    {
+        return "TRACE";
+    }
+    else if (level == wolkabout::LogLevel::OFF)
+    {
+        return "OFF";
+    }
+    else
+    {
+        return "";
+    }
+}
+// returns LogLevel and if string is valid LogLevel
+std::pair<wolkabout::LogLevel, bool> fromString(std::string in)
+{
+    if (in == "INFO")
+    {
+        return std::make_pair(wolkabout::LogLevel::INFO, true);
+    }
+    else if (in == "WARN")
+    {
+        return std::make_pair(wolkabout::LogLevel::WARN, true);
+    }
+    else if (in == "ERROR")
+    {
+        return std::make_pair(wolkabout::LogLevel::ERROR, true);
+    }
+    else if (in == "DEBUG")
+    {
+        return std::make_pair(wolkabout::LogLevel::DEBUG, true);
+    }
+    else if (in == "TRACE")
+    {
+        return std::make_pair(wolkabout::LogLevel::TRACE, true);
+    }
+    else if (in == "OFF")
+    {
+        return std::make_pair(wolkabout::LogLevel::OFF, true);
+    }
+    else
+    {
+        return std::make_pair(wolkabout::LogLevel::INFO, false);
+    }
+}
+// Class for handling recieved data from feed
+class DeviceDataChangeHandler : public wolkabout::connect::FeedUpdateHandler
+{
+public:
+    DeviceDataChangeHandler(std::string log, bool valid) : m_previousLog(log), m_isLogValid(valid) {}
+
+    void handleUpdate(const std::string& deviceKey,
+                      const std::map<std::uint64_t, std::vector<wolkabout::Reading>>& readings) override
+    {
+        // Go through all the timestamps
+        for (const auto& pair : readings)
+        {
+            LOG(DEBUG) << "Received feed information for time: " << pair.first;
+
+            // Take the readings, and apply them
+            for (const auto& reading : pair.second)
+            {
+                LOG(DEBUG) << "Received feed information for reference '" << reading.getReference() << "'.";
+
+                // Lock the mutex
+                std::lock_guard<std::mutex> lock{mutex};
+
+                // Check the reference on the readings
+                if (reading.getReference() == "log")
+                {
+                    auto in = fromString(reading.getStringValue());
+                    if (in.second)
+                    {
+                        wolkabout::Logger::getInstance().setLevel(in.first);
+                        m_previousLog = toString(in.first);
+                        LOG(INFO) << "Log has been updated";
+                    }
+                    else
+                    {
+                        LOG(INFO) << "Unknown log value";
+                        m_isLogValid = false;
+                    }
+                }
+            }
+
+            // Notify the condition variable
+            conditionVariable.notify_one();
```

Comment:
Since the conditionVariable is here already, perhaps use it to make this solution a bit more elegant (notify thread to send actual value of LogLevel to platform).
https://en.cppreference.com/w/cpp/thread/condition_variable.html